element_interface/intan_loader.py

```diff
@@ -0,0 +1,298 @@
+                    elif signal_type == 2:
+                        raise Exception("Wrong signal type for the rhs format")
+                        # header['supply_voltage_channels'].append(new_channel)
+                    elif signal_type == 3:
+                        header["board_adc_channels"].append(new_channel)
+                    elif signal_type == 4:
+                        header["board_dac_channels"].append(new_channel)
+                    elif signal_type == 5:
+                        header["board_dig_in_channels"].append(new_channel)
+                    elif signal_type == 6:
+                        header["board_dig_out_channels"].append(new_channel)
+                    else:
+                        raise Exception("Unknown channel type.")
+
+    # Summarize contents of data file.
+    header["num_amplifier_channels"] = len(header["amplifier_channels"])
+    header["num_board_adc_channels"] = len(header["board_adc_channels"])
+    header["num_board_dac_channels"] = len(header["board_dac_channels"])
+    header["num_board_dig_in_channels"] = len(header["board_dig_in_channels"])
+    header["num_board_dig_out_channels"] = len(header["board_dig_out_channels"])
+
+    return header
+
+
+def load_rhs(folder: str, file_expr: str = "*"):
+    """Load rhs data
+
+    Example:
+        # Read data
+        >>> rhs_data = load_rhs("/home/inbox/organoids21/032520_US_885kHz_sham", file_expr="amp*dat")
+
+        # Plot data
+        >>> import matplotlib.pyplot as plt
+        >>> plt.plot(rhs_data["time"], rhs_data["recordings"]["amp-B-000.dat"])
+        >>> plt.xlabel("Time (s)")
+        >>> plt.ylabel("Reading")
+        >>> plt.show()
+
+    Args:
+        folder (str): Folder that contains info.rhs, time.dat, and *.dat files
+        file_expr (str): pattern matching of file names to be read. Defaults to "*" (read all files).
+
+    Returns:
+        rhs_data (dict): RHS data.
+            rhs_data["header"] (dict): Header.
+            rhs_data["recordings"] (dict): Readings from various files
+            rhs_data["timestamps"] (np.array_like): Relative timestamps in seconds.
+    """
+
+    rhs_data = {}
+
+    # Get header
+    header_filepath = next(Path(folder).glob("info.rhs"))
+    with open(header_filepath, "rb") as fid:
+        rhs_data["header"] = _read_header(fid)
+
+    # Get timestamps
+    time_file = next(Path(folder).glob("time.dat"))
+
+    rhs_data["timestamps"] = (
+        np.memmap(time_file, dtype=np.int32)
+        / rhs_data["header"]["frequency_parameters"]["amplifier_sample_rate"]
+    )
+
+    # Get data files
+    file_paths = Path(folder).glob(file_expr)
+
+    exclude_list = ["time", "info", "Zone.Identifier"]
+
+    file_paths = [
+        file
+        for file in file_paths
+        if not any(string in file.as_posix() for string in exclude_list)
+    ]
+
+    # Get recording data
+    rhs_data["recordings"] = {}
+
+    if len(file_paths):
```

Comment:
```suggestion
```
 I don't this is needed..  If it's empty, it won't enter the loop.

Reply:
You are right.